tests/compiletests.rs

```diff
@@ -0,0 +1,13 @@
+extern crate compiletest_rs as compiletest;
+
+use std::path::PathBuf;
+
+#[test]
+fn odd_buffer_compile_fail() {
+    // Test that `BufEncoder` fails to compile when the capacity is odd.
+    let mut config = compiletest::Config::default();
+    config.mode = "compile-fail".parse().expect("Invalid mode");
+    config.src_base = PathBuf::from("tests/compiletest");
+    config.target_rustcflags = Some("-L target/debug/deps".to_string());
+    compiletest::run_tests(&config);
+}
```

Comment:
Looks quite complicated. Why not simply have a new CI step to run `if cargo check --something; then exit 1; fi`?

Reply:
I wasn't sure about adding a CI step just for this one, should be simple, test.  But it does work, and there might be reasons in the future to add to the compile tests?

Reply:
I think it's probably worthwhile to do this in a general way (modifying CI). If we had the means to add compile-tests I'll bet we'd add a lot of them.

Reply:
I prefer not to take dev dependency if the same can be done in CI.